src/runtime/entitlements-manager.js

```diff
@@ -356,75 +402,157 @@ export class EntitlementsManager {
    */
-  maybeShowToast_(entitlements) {
-    const entitlement = entitlements.getEntitlementForThis();
-    if (!entitlement) {
-      return Promise.resolve();
-    }
+  maybeShowToast_(entitlement) {
     // Check if storage bit is set. It's only set by the `Entitlements.ack`
     // method.
     return this.storage_.get(TOAST_STORAGE_KEY).then((value) => {
-      if (value == '1') {
-        // Already shown;
+      const toastWasShown = value === '1';
+      if (toastWasShown) {
         return;
       }
-      if (entitlement) {
-        this.showToast_(entitlement);
-      }
+
+      // Show toast.
+      const source = entitlement.source || 'google';
+      return new Toast(
+        this.deps_,
+        feUrl('/toastiframe'),
+        feArgs({
+          'publicationId': this.publicationId_,
+          'source': source,
+        })
+      ).open();
     });
   }
 
   /**
-   * @param {!Entitlement} entitlement
+   * @param {!Entitlements} entitlements
    * @private
    */
-  showToast_(entitlement) {
-    const source = entitlement.source || 'google';
-    return new Toast(
-      this.deps_,
-      feUrl('/toastiframe'),
-      feArgs({
-        'publicationId': this.publicationId_,
-        'source': source,
-      })
-    ).open();
+  ack_(entitlements) {
+    if (entitlements.getEntitlementForThis()) {
+      this.setToastShown(true);
+    }
   }
 
   /**
    * @param {!Entitlements} entitlements
    * @private
    */
-  ack_(entitlements) {
-    if (entitlements.getEntitlementForThis()) {
-      this.setToastShown(true);
+  consume_(entitlements) {
+    if (entitlements.enablesThisWithGoogleMetering()) {
+      // TODO: Open Metering Prompt.
+      const activityIframeView_ = new ActivityIframeView(
+        this.win_,
+        this.deps_.activities(),
+        feUrl('/contributionsiframe'),
+        feArgs({
+          'productId': this.deps_.pageConfig().getProductId(),
+          'publicationId': this.deps_.pageConfig().getPublicationId(),
+          'productType': ProductType.UI_CONTRIBUTION,
+          'list': 'default',
+          'skus': null,
+          'isClosable': true,
+        }),
+        /* shouldFadeBody */ true
+      );
+      activityIframeView_.onCancel(() => {
+        this.sendPingback_(entitlements);
+      });
+      return this.deps_.dialogManager().openView(activityIframeView_);
     }
   }
 
   /**
-   * @param {?string=} encryptedDocumentKey
+   * @param {!GetEntitlementsParams=} params
    * @return {!Promise<!Entitlements>}
    * @private
    */
-  fetch_(encryptedDocumentKey) {
-    let url =
-      '/publication/' +
-      encodeURIComponent(this.publicationId_) +
-      '/entitlements';
-    if (encryptedDocumentKey) {
-      url += '?crypt=' + encodeURIComponent(encryptedDocumentKey);
-    }
-    return this.fetcher_
-      .fetchCredentialedJson(serviceUrl(url))
-      .then((json) => this.parseEntitlements(json));
+  fetch_(params) {
+    return hash(getCanonicalUrl(this.deps_.doc()))
```

Comment:
Better to build a new params object than mess the the one passed - a) the caller might be surprised their object got trashed and b) they may have crap in there we don't care about.  If you make a new object you know exactly what's in it.

Reply:
Good call, the code is now creating a new object